### PR TITLE
DNN-10252: only allow attach files in current user folder.

### PR DESCRIPTION
**Pull request by zyhfish**

### CODE REVIEW

**Comment by galatrash-at-dnn** [comment]
@zyhfish , you should add some tests both positive and negative) for your code changes.

File: DNN Platform/Modules/Journal/ServicesController.cs [hunk 1]
```diff
@@ -119,14 +119,31 @@ public HttpResponseMessage Create(CreateDTO postData)
                     postData.ProfileId = UserInfo.UserID;
                 }
 
+                if (postData.ProfileId != UserInfo.UserID)
+                {
+                    var profileUser = UserController.Instance.GetUser(PortalSettings.PortalId, postData.ProfileId);
+                    if (profileUser == null || (!UserInfo.IsInRole(PortalSettings.AdministratorRoleName) && !Utilities.AreFriends(profileUser, UserInfo)))
+                    {
+                        throw new ArgumentException("you have no permission to post journal on current profile page");
```
Comment (galatrash-at-dnn) [inline DNN Platform/Modules/Journal/ServicesController.cs]: Use ``Localization.GetString(...)`` please

File: DNN Platform/Modules/Journal/ServicesController.cs [hunk 2]
```diff
@@ -119,14 +119,31 @@ public HttpResponseMessage Create(CreateDTO postData)
...
                     postData.ProfileId = -1;
 
                     RoleInfo roleInfo = RoleController.Instance.GetRoleById(ActiveModule.OwnerPortalID, postData.GroupId);
-                    if (roleInfo != null && !roleInfo.IsPublic)
+                    if (roleInfo != null)
                     {
-                        postData.SecuritySet = "R";
+                        if (!UserInfo.IsInRole(PortalSettings.AdministratorRoleName) && !UserInfo.IsInRole(roleInfo.RoleName))
+                        {
+                            throw new ArgumentException("you have no permission to post journal on current group.");
```
Comment (galatrash-at-dnn) [inline DNN Platform/Modules/Journal/ServicesController.cs]: Use ``Localization.GetString(...)`` please

File: DNN Platform/Modules/Journal/ServicesController.cs [hunk 3]
```diff
@@ -119,14 +119,31 @@ public HttpResponseMessage Create(CreateDTO postData)
                     postData.ProfileId = UserInfo.UserID;
```
Comment (galatrash-at-dnn) [inline DNN Platform/Modules/Journal/ServicesController.cs]: You need to remove the ``if`` condition of this block to become basically ``postData.ProfileId = UserInfo.UserID;`` instead of ``if(..) { postData.ProfileId = UserInfo.UserID; }

File: DNN Platform/Modules/Journal/ServicesController.cs [hunk 4]
```diff
@@ -119,14 +119,31 @@ public HttpResponseMessage Create(CreateDTO postData)
...
+                    var profileUser = UserController.Instance.GetUser(PortalSettings.PortalId, postData.ProfileId);
+                    if (profileUser == null || (!UserInfo.IsInRole(PortalSettings.AdministratorRoleName) && !Utilities.AreFriends(profileUser, UserInfo)))
+                    {
+                        throw new ArgumentException("you have no permission to post journal on current profile page");
+                    }
+                }
+
                 if (postData.GroupId > 0)
                 {
                     postData.ProfileId = -1;
```
Comment (galatrash-at-dnn) [inline DNN Platform/Modules/Journal/ServicesController.cs]: I know this is an old line of code, but why is it is needed? don't we keep the UserID when posting to a group?

File: DNN Platform/Modules/Journal/ServicesController.cs (hunk 3 above)
Comment (zyhfish) [reply DNN Platform/Modules/Journal/ServicesController.cs]: The if condition can't be removed, as have scenario user can post journal on friend's profile page.
